apps/benchmarking-test-app/src/benchmarks/irEcsBenchmark.ts

```diff
@@ -0,0 +1,83 @@
+import {
+  AnimationSystemGroup,
+  createEngine,
+  createEntity,
+  defineComponent,
+  defineQuery,
+  defineSystem,
+  destroyEngine,
+  Entity,
+  executeSystems,
+  hasComponent,
+  removeComponent,
+  setComponent,
+} from "@ir-engine/ecs";
+
+const ITERATIONS = 5000;
+
+const mockDeltaMillis = 1000 / 60;
+
+export const irEcsBenchmark = async () => {
+  await new Promise((resolve) => setTimeout(resolve, 10));
+
+  const MockComponent = setup();
+
+  createEngine();
+
+  const entity = createEntity();
+
+  for (let i = 0; i < ITERATIONS; i++) {
+    setComponent(entity, MockComponent, { mockValue: i });
+    hasComponent(entity, MockComponent);
```

Comment:
should also add a getComponent as well since it's another of the most common component functions.

Reply:
Updated! Also added getOptionalComponent after the removeComponent call